test/mocha/schedule.js

```diff
@@ -44,5 +44,16 @@ describe("schedule", function () {
                 assert.fail();
             });
         });
+
+        describe("Promise.setTimeoutScheduler", function() {
+            it("should throw for non function", function() {
+                try {
+                    Promise.setTimeoutScheduler({});
+                } catch (e) {
+                    return Promise.resolve();
+                }
+                assert.fail();
```

Comment:
this test does succeed (returns) when catch cause is called. Are you suggesting that we remove the ```assert.fail()```? I think assert.fail is good to ensure that test will fail if catch wasnt called.